Add extension method for easier `ResourceManagementOptions` registration (Lombiq Technologies: OCORE-208) (#17095)
OrchardCMS/OrchardCore: src/docs/releases/3.0.0.md

Many type commonly used by classes can be `sealed`, which improves runtime performance. While it's not mandatory, we recommend that you consider applying this improvement to your own extensions as well. We've implemented this enhancement in pull request [#16897](https://github.com/OrchardCMS/OrchardCore/pull/16897).

## Change Log

> 

Reviewer: I would add these under miscellaneous section or add these notes under "Resources" section.

Author: Created a Miscellaneous section.

### New Extension Method for Adding `IConfigureOptions<ResourceManagementOptions>` Implementations

When adding an `IConfigureOptions<ResourceManagementOptions>` implementation, used to add static resources and commonly named `ResourceManagementOptionsConfiguration`, you previously had to do the following in `Startup` classes:

```csharp
services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
```

To simplify this, we introduced a new extension method to do the same in a shorter form:

```csharp

> services.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();

Reviewer 2: ```suggestion
services.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

```

> You can utilize this in your codebase by doing a search for the `AddTransient<IConfigureOptions<ResourceManagementOptions>, (.+)>\(\)` regex pattern and replacing it with `AddResourceConfiguration<$1>()`. Projects using this have to reference the `OrchardCore.ResourceManagement` package.

Reviewer: Or you. An right click on IConfigureOptions and select show all references :)

Author: That'll show you where the affected code lines are, but then you'd need to manually change them. With a regex (and obviously also checking the git diff, and testing the code) you get it done in 5 seconds :).

> You can utilize this in your codebase by doing a search for the `AddTransient<IConfigureOptions<ResourceManagementOptions>, (.+)>\(\)` regex pattern and replacing it with `AddResourceManagementOptionsConfiguration<$1>()`. Projects using this have to reference the `OrchardCore.ResourceManagement` package.

Reviewer 2: ```suggestion
You can utilize this in your codebase by doing a search for the `AddTransient<IConfigureOptions<ResourceManagementOptions>, (.+)>\(\)` regex pattern and replacing it with `ConfigureResourceManagementOptions<$1>()`. Projects using this have to reference the `OrchardCore.ResourceManagement` package.
```

Many type commonly used by classes can be `sealed`, which improves runtime performance. While it's not mandatory, we recommend that you consider applying this improvement to your own extensions as well. We've implemented this enhancement in pull request [#16897](https://github.com/OrchardCMS/OrchardCore/pull/16897).

## Change Log

## Miscellaneous

### New Extension Method for Adding `IConfigureOptions<ResourceManagementOptions>` Implementations

> When adding an `IConfigureOptions<ResourceManagementOptions>` implementation, used to add static resources and commonly named `ResourceManagementOptionsConfiguration`, you previously had to do the following in `Startup` classes:

Reviewer 2: ```suggestion
When adding an `IConfigureOptions<ResourceManagementOptions>` implementation, used to add static resources and commonly named `ResourceManagementOptionsConfiguration`, you previously had to do the following in the `Startup` classes:
```

```csharp
services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
```

To simplify this, we introduced a new extension method to do the same in a shorter form:

```csharp
services.AddResourceConfiguration<ResourceManagementOptionsConfiguration>();
```

> You can utilize this in your codebase by doing a search for the `AddTransient<IConfigureOptions<ResourceManagementOptions>, (.+)>\(\)` regex pattern and replacing it with `AddResourceConfiguration<$1>()`. Projects using this have to reference the `OrchardCore.ResourceManagement` package.

Reviewer 2: ```suggestion
You can utilize this in your codebase by searching the `AddTransient<IConfigureOptions<ResourceManagementOptions>, (.+)>\(\)` regex pattern and replacing it with `AddResourceConfiguration<$1>()`. Projects using this have to reference the `OrchardCore.ResourceManagement` package.
```
